### DISCUSSION TITLE
Why resetting the SMC and NVRAM don't work after uninstalling Al Dente?

### DISCUSSION THREAD

**Discussion by metebyte** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#983] [Q&A]
Hi there Al Dente community,

As I mentioned on the title above, Al Dente still affects the charge process even after completely uninstalling it. Resetting the SMC and NVRAM work but only per boot. I mean I have to reset the SMC at every time I boot. Otherwise it doesn't work. Has anyone experienced something like this issue?

**Comment by MatthiasKerbl** [comment]
Hi @metebyte,

This should not be the case. In which way is AlDente still affecting your charging behavior after you uninstalled it and reset the SMC and NVRAM? Which MacBook model and macOS version are you using?

Best,
Matthias

**Reply by metebyte** [reply]
Hello @MatthiasKerbl ,

First of all, thank you for your response. And yes, strangely enough, the charging function of my computer is still affected. My computer doesn't charge unless I reset the SMC, and I have to reset the SMC every time I restart. Even after performing a clean macOS installation, I still encounter this issue. Lastly, I am using a MacBook Pro Model A1708. The macOS installed is the Ventura version using OCLP. I will eagerly await your response.

**Reply by metebyte** [reply]
By the way, I would like to provide a few more additional details for assistance. The battery cycle on my MacBook was 320, and its condition appeared to be normal. However, I still replaced it with another OEM battery that had a cycle count of 42. The result is still the same: the charging function does not work in any way without performing an SMC reset. I mean even replacing the battery doesn't allow MacBook to charge properly.

**Reply by MatthiasKerbl** [reply]
Hi @metebyte,

Thank you for the detailed description. To be honest, I do not know what exactly the cause of this issue is. If AlDente is properly uninstalled by following our [uninstallation guide](https://apphousekitchen.com/how-to-uninstall-aldente-pro/) and an SMC  and NVRAM reset are performed, all changes of AlDente are gone from the system.
One option you could try is to install AlDente again and perform the uninstallation following our [guide](https://apphousekitchen.com/how-to-uninstall-aldente-pro/).
Furthermore, it is of course also a possibility that this issue is completely independent of AlDente and is caused by a hardware issue.

Best,
Matthias

**Reply by metebyte** [reply]
Hi @MatthiasKerbl,

Thank you for your response. I did the installation and uninstallation again but no luck. Is there any log provider or something related to Apple's Power Management Unit that I can check what's going on?

<img width="411" alt="Screenshot 2023-07-04 at 01 35 04" src="https://github.com/davidwernhart/AlDente-Charge-Limiter/assets/92409276/86f262a2-f0a4-4fcf-840b-424268c44d69">

**Reply by metebyte** [reply]
This is an update regarding the situation.

Today, I installed Al Dente on my other Intel-based MacBook Pro. Previously, I hadn't installed any third-party battery or power management programs on this device. Therefore, after a quick installation, I performed a calibration process and consumed some power to see If the charge limit feature is working. Then, I proceeded with the removal process. I used the device for 7 hours, during which the battery was consumed and subsequently recharged. However, the result was exactly the same. The charging function does not work as it should unless an SMC reset is performed. Consequently, I believe that Al Dente Pro is causing harm to Intel-based Macs. I hope to receive clarification on this matter soon.

**Reply by MatthiasKerbl** [reply]
Hi @metebyte,

Thank you for continuously testing and reporting about this issue. We ourselves and our users have been running AlDente Pro on Intel MacBooks for years and have never encountered this issue or rather you are the first one to report this issue. Furthermore, we have just tested this exact scenario on a 13" MacBook Pro 2017 and we are not able to reproduce your issue at all, everything is working as excepted.
Is it possible that your charging cable or charger itself has a defect? Do you have different cables and chargers available to test this hypothesis?
Furthermore, do you have Apple's Optimized Battery Charging deactivated?

Best,
Matthias

**Reply by metebyte** [reply]
Hi @MatthiasKerbl,

Thank you for your response, and I'd like to add that all the thoughts I've shared so far are purely theoretical. However, since I couldn't obtain any concrete data, I asked for help from you regarding this matter. As a final and additional update, I should note that the same results were encountered in all scenarios, and I held Al Dente responsible for the issue. Both Apple MNF72TU/A 61W and Apple MNF82TU/A 87W power adapters yielded the same results. Nevertheless, I believe I managed to temporarily resolve the problem on two separate MacBook Pro devices. I accessed the keys with SMCKit, and now I can charge without performing an SMC reset before each boot. And yes, Apple's Optimized Battery Charging was disabled.

**Reply by metebyte** [reply]
My final theory is that the battery might have been inaccurately calibrated without any warning from Al Dente. Because ultimately, the lack of synchronization between the hardware-communicated data and the values on the software side could potentially prevent macOS from initiating the charging process.
